test/integration/spec/bandwidthprofile.js

```diff
@@ -0,0 +1,255 @@
+/* eslint-disable no-undefined */
+'use strict';
+
+const assert = require('assert');
+const connect = require('../../../lib/connect');
+const { audio: createLocalAudioTrack, video: createLocalVideoTrack } = require('../../../lib/createlocaltrack');
+const defaults = require('../../lib/defaults');
+const { createRoom, completeRoom } = require('../../lib/rest');
+const getToken = require('../../lib/token');
+
+const {
+  capitalize,
+  combinationContext,
+  createSyntheticAudioStreamTrack,
+  dominantSpeakerChanged,
+  randomName,
+  setup,
+  smallVideoConstraints,
+  tracksSubscribed,
+  trackSwitchedOff,
+  trackSwitchedOn,
+  waitFor
+} = require('../../lib/util');
+
+const { trackPriority: { PRIORITY_HIGH, PRIORITY_LOW, PRIORITY_STANDARD } } = require('../../../lib/util/constants');
+const { trackSwitchOffMode: { MODE_DISABLED, MODE_DETECTED, MODE_PREDICTED } } = require('../../../lib/util/constants');
+
+describe('Bandwidth Management', function() {
+  // eslint-disable-next-line no-invalid-this
+  this.timeout(120 * 1000);
+
+  if (defaults.topology !== 'peer-to-peer') {
+    describe('bandwidthProfile.video', () => {
+      combinationContext([
+        [
+          [{ maxSubscriptionBitrate: 400 }, { maxTracks: 1 }],
+          ({ maxSubscriptionBitrate, maxTracks }) => maxSubscriptionBitrate
+            ? `.maxSubscriptionBitrate = ${maxSubscriptionBitrate}`
+            : `.maxTracks = ${maxTracks}`
+        ],
+        [
+          [PRIORITY_LOW, PRIORITY_STANDARD, PRIORITY_HIGH],
+          x => `.dominantSpeakerPriority = "${x}"`
+        ],
+        [
+          [PRIORITY_LOW, PRIORITY_STANDARD, PRIORITY_HIGH],
+          x => `and the publish priority of the Dominant Speaker's LocalVideoTrack is "${x}"`
+        ],
+        [
+          [PRIORITY_LOW, PRIORITY_STANDARD, PRIORITY_HIGH],
+          x => `and the publish priority of the Passive Speaker's LocalVideoTrack is "${x}"`
+        ]
+      ], ([trackLimitOptions, dominantSpeakerPriority, dominantSpeakerPublishPriority, passiveSpeakerPublishPriority]) => {
+        const priorityRanks = {
+          [PRIORITY_HIGH]: 1,
+          [PRIORITY_STANDARD]: 2,
+          [PRIORITY_LOW]: 3
+        };
+
+        // NOTE(mmalavalli): Since "dominantSpeakerPriority" only upgrades the publish priority of the Dominant Speaker's
```

Comment:
dominantSpeakerPriority is a subscriber-side priority and It should not change publishPriority. isn't it?

Reply:
yes, it does not change the publish side priority. I will edit the comment.